2023-09-18 17:47:45.547743: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2023-09-18 17:47:45.547921: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2023-09-18 17:47:53,473 | app.py:151 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2023-09-18 17:47:53,553 | app.py:172 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2023-09-18 17:47:53,553 | server.py:86 | Initializing global parameters
INFO flwr 2023-09-18 17:47:53,553 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2023-09-18 17:47:53,866 | server.py:277 | Received initial parameters from one random client
INFO flwr 2023-09-18 17:47:53,866 | server.py:88 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:47:53,882 | server.py:95 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.44}
INFO flwr 2023-09-18 17:47:53,883 | server.py:101 | FL starting
DEBUG flwr 2023-09-18 17:47:54,251 | server.py:223 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2023-09-18 17:47:54,279 | server.py:237 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2023-09-18 17:47:54,282 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:47:54,295 | server.py:125 | fit progress: (1, 0.4107493600604092, {'accuracy': 0.7833333333333333}, 0.4123044999999994)
DEBUG flwr 2023-09-18 17:47:54,295 | server.py:173 | evaluate_round 1: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:47:54,326 | server.py:187 | evaluate_round 1 received 4 results and 0 failures
WARNING flwr 2023-09-18 17:47:54,326 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2023-09-18 17:47:54,327 | server.py:223 | fit_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:47:54,357 | server.py:237 | fit_round 2 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:47:54,370 | server.py:125 | fit progress: (2, 0.420067656126209, {'accuracy': 0.7833333333333333}, 0.4876454999999993)
DEBUG flwr 2023-09-18 17:47:54,371 | server.py:173 | evaluate_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:47:54,411 | server.py:187 | evaluate_round 2 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:47:54,412 | server.py:223 | fit_round 3: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:47:54,453 | server.py:237 | fit_round 3 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:47:54,465 | server.py:125 | fit progress: (3, 0.42059150709451476, {'accuracy': 0.7833333333333333}, 0.5819101)
DEBUG flwr 2023-09-18 17:47:54,466 | server.py:173 | evaluate_round 3: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:47:54,490 | server.py:187 | evaluate_round 3 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:47:54,490 | server.py:223 | fit_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:47:54,534 | server.py:237 | fit_round 4 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:47:54,547 | server.py:125 | fit progress: (4, 0.4205684156369666, {'accuracy': 0.7833333333333333}, 0.6643589999999993)
DEBUG flwr 2023-09-18 17:47:54,547 | server.py:173 | evaluate_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:47:54,578 | server.py:187 | evaluate_round 4 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:47:54,578 | server.py:223 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:47:54,622 | server.py:237 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:47:54,635 | server.py:125 | fit progress: (5, 0.4205673299707892, {'accuracy': 0.7833333333333333}, 0.7516914999999997)
DEBUG flwr 2023-09-18 17:47:54,635 | server.py:173 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:47:54,670 | server.py:187 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:47:54,670 | server.py:223 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:47:54,713 | server.py:237 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:47:54,726 | server.py:125 | fit progress: (6, 0.4205672140981427, {'accuracy': 0.7833333333333333}, 0.8439010000000007)
DEBUG flwr 2023-09-18 17:47:54,727 | server.py:173 | evaluate_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:47:54,774 | server.py:187 | evaluate_round 6 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:47:54,774 | server.py:223 | fit_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:47:54,826 | server.py:237 | fit_round 7 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:47:54,841 | server.py:125 | fit progress: (7, 0.41541556770075255, {'accuracy': 0.79}, 0.9581110000000006)
DEBUG flwr 2023-09-18 17:47:54,841 | server.py:173 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:47:54,882 | server.py:187 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:47:54,882 | server.py:223 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:47:54,935 | server.py:237 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:47:54,947 | server.py:125 | fit progress: (8, 0.41544167840823876, {'accuracy': 0.79}, 1.0648441999999996)
DEBUG flwr 2023-09-18 17:47:54,947 | server.py:173 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:47:54,988 | server.py:187 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:47:54,988 | server.py:223 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:47:55,034 | server.py:237 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:47:55,048 | server.py:125 | fit progress: (9, 0.41544099210226826, {'accuracy': 0.79}, 1.1649369000000007)
DEBUG flwr 2023-09-18 17:47:55,048 | server.py:173 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:47:55,080 | server.py:187 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:47:55,080 | server.py:223 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:47:55,120 | server.py:237 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:47:55,133 | server.py:125 | fit progress: (10, 0.4154410025525948, {'accuracy': 0.79}, 1.250217899999999)
DEBUG flwr 2023-09-18 17:47:55,134 | server.py:173 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:47:55,165 | server.py:187 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2023-09-18 17:47:55,165 | server.py:147 | FL finished in 1.2822166999999993
INFO flwr 2023-09-18 17:47:55,166 | app.py:218 | app_fit: losses_distributed [(1, 0.3968057068330901), (2, 0.39158221759966444), (3, 0.3996546850318), (4, 0.39965453885850455), (5, 0.3996545246669224), (6, 0.40390479138919283), (7, 0.40328383701188225), (8, 0.4032840664897646), (9, 0.40328404222215924), (10, 0.40328404222215924)]
INFO flwr 2023-09-18 17:47:55,166 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2023-09-18 17:47:55,166 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2023-09-18 17:47:55,166 | app.py:221 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.4107493600604092), (2, 0.420067656126209), (3, 0.42059150709451476), (4, 0.4205684156369666), (5, 0.4205673299707892), (6, 0.4205672140981427), (7, 0.41541556770075255), (8, 0.41544167840823876), (9, 0.41544099210226826), (10, 0.4154410025525948)]
INFO flwr 2023-09-18 17:47:55,166 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.44), (1, 0.7833333333333333), (2, 0.7833333333333333), (3, 0.7833333333333333), (4, 0.7833333333333333), (5, 0.7833333333333333), (6, 0.7833333333333333), (7, 0.79), (8, 0.79), (9, 0.79), (10, 0.79)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.87509064,  1.1711375 ,  0.86250393,  0.28897747,  0.10810526,
         0.03296164, -0.02861145,  0.09073561]]), array([-0.17850804])]
[array([[-1.87509064,  1.1711375 ,  0.86250393,  0.28897747,  0.10810526,
         0.03296164, -0.02861145,  0.09073561]]), array([-0.17850804])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.08682072,  1.05705214,  0.80986097,  0.4021422 ,  0.17344002,
         0.11447449,  0.03909871,  0.04399173]]), array([-0.06830801])]
[array([[-2.08682072,  1.05705214,  0.80986097,  0.4021422 ,  0.17344002,
         0.11447449,  0.03909871,  0.04399173]]), array([-0.06830801])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03413602,  1.00819445,  0.78754046,  0.40456049,  0.16595092,
         0.1156636 ,  0.04025409,  0.06246675]]), array([-0.06905558])]
[array([[-2.03413602,  1.00819445,  0.78754046,  0.40456049,  0.16595092,
         0.1156636 ,  0.04025409,  0.06246675]]), array([-0.06905558])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03412745,  1.00813566,  0.78739726,  0.40427636,  0.16587923,
         0.11535988,  0.0400776 ,  0.0625186 ]]), array([-0.06782202])]
[array([[-2.03412745,  1.00813566,  0.78739726,  0.40427636,  0.16587923,
         0.11535988,  0.0400776 ,  0.0625186 ]]), array([-0.06782202])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03412585,  1.00813876,  0.78741951,  0.40424923,  0.16588684,
         0.11533145,  0.04007497,  0.06252374]]), array([-0.06778163])]
[array([[-2.03412585,  1.00813876,  0.78741951,  0.40424923,  0.16588684,
         0.11533145,  0.04007497,  0.06252374]]), array([-0.06778163])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03412598,  1.00813795,  0.78741949,  0.40424755,  0.16588616,
         0.11532942,  0.04007412,  0.06252315]]), array([-0.06777301])]
[array([[-2.03412598,  1.00813795,  0.78741949,  0.40424755,  0.16588616,
         0.11532942,  0.04007412,  0.06252315]]), array([-0.06777301])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.00738065,  0.98209598,  0.80373133,  0.37337297,  0.11798318,
         0.10974145, -0.03485006,  0.05931255]]), array([0.08400732])]
[array([[-2.00738065,  0.98209598,  0.80373133,  0.37337297,  0.11798318,
         0.10974145, -0.03485006,  0.05931255]]), array([0.08400732])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.00726552,  0.9820736 ,  0.80403718,  0.37366002,  0.1180508 ,
         0.11004964, -0.03463373,  0.05925138]]), array([0.08257216])]
[array([[-2.00726552,  0.9820736 ,  0.80403718,  0.37366002,  0.1180508 ,
         0.11004964, -0.03463373,  0.05925138]]), array([0.08257216])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.00728082,  0.98208916,  0.80403868,  0.37365834,  0.11803836,
         0.11004085, -0.03463254,  0.05924284]]), array([0.08260944])]
[array([[-2.00728082,  0.98208916,  0.80403868,  0.37365834,  0.11803836,
         0.11004085, -0.03463254,  0.05924284]]), array([0.08260944])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.00728021,  0.98208876,  0.80403854,  0.37365836,  0.11803872,
         0.11004104, -0.03463273,  0.05924299]]), array([0.08260861])]
[array([[-2.00728021,  0.98208876,  0.80403854,  0.37365836,  0.11803872,
         0.11004104, -0.03463273,  0.05924299]]), array([0.08260861])]
>>>>>>>>>>>>>run time:
0.59375s
